### ISSUE TITLE
Godot 4 rc2: shortcut for line comment toggling, bug if cursor at beginning of line

### ISSUE THREAD

**Issue by Juuliuus** [godotengine/godot#73367]
### Godot version

4.0 rc2

### System information

Linux Ubuntu

### Issue description

If the cursor is at the beginning of the line and then using the comment line toggle shortcut the # is placed AFTER the cursor instead of before it. This behavior has changed since rc1 I believe because I would have noticed it.

The expected behavior is that the # will be placed before the cursor.

I often go to the beginning of a blank line, invoke the toggle comment shortcut, and then start typing. With this current behavior it starts pushing the # which is unexpected and more work while then I have to move the cursor to after the #.

### Steps to reproduce

On Linux version, Godot 4 rc2 place your cursor on a blank empty line, or at the beginning of a code line.

Invoke line comment toggle shortcut, the described above behavior occurs with the # being placed after the cursor instead of in front of it.

### Minimal reproduction project

<no script>

**Comment by akien-mga** [comment]
I can reproduce the issue, and confirm that it changed in RC 2.

Probably caused by #72671, CC @aXu-AP @Paulb23 (didn't test to confirm that hunch though).

**Comment by aXu-AP** [comment]
Yes, this is from my rewrite. I didn't notice the distinction of selection being at the start of the line vs only caret being there. It's an easy fix, I'll try do it tomorrow (too late to wait compiling Godot 😅).
